lib/csv/parser.rb

```diff
@@ -409,13 +409,7 @@ def parse(&block)
 
       begin
         @scanner ||= build_scanner
-        if quote_character.nil?
-          parse_no_quote(&block)
-        elsif @need_robust_parsing
-          parse_quotable_robust(&block)
-        else
-          parse_quotable_loose(&block)
-        end
+        @parse_method.call(&block)
```

Comment:
Could you use `:parse_XXX` and `__send__` instead of `method(:parse_XXX)` and `call`?

```suggestion
        __send__(@parse_method, &block)
```